pkg/http/download_test.go

```diff
@@ -0,0 +1,66 @@
+package http
+
+import (
+	"context"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestDownloadFile(t *testing.T) {
+	tests := []struct {
+		name        string
+		ctx         context.Context
+		url         string
+		path        string
+		expectedErr string
+	}{
+		{
+			name:        "Nil context",
+			expectedErr: "creating request: net/http: nil Context",
+		},
+		{
+			name:        "Invalid URL",
+			ctx:         context.Background(),
+			url:         "invalid-url",
+			expectedErr: "executing request: Get \"invalid-url\": unsupported protocol scheme \"\"",
+		},
+		{
+			name:        "Unexpected status",
+			ctx:         context.Background(),
+			url:         "https://github.com/suse-edge/eib",
+			expectedErr: "unexpected status code: 404",
+		},
+		{
+			name:        "Error creating file",
+			ctx:         context.Background(),
+			url:         "https://github.com/suse-edge/edge-image-builder",
```

Comment:
This is tricky since I'm generally not a fan of unit tests making a network call. Is there any sort of way in Go to differentiate between "unit" and "integration" tests, such that we can only have this test run in CI but not by default locally? The answer may be that I need to do more than `go test ./...` locally. Just curious if you had any experience with this sort of test differentiation.

Reply:
I considered these lightweight enough not to label as integration tests. However, it is indeed possible and actually very easy to implement the separation via "build tags". Here is a short [blog post](https://mickey.dev/posts/go-build-tags-testing/) showcasing how it's done.

I'll come back to this once I figure out how much of the functionality here will be used for the Kubernetes combustion components.

Reply:
Works for me 👍🏼